doc/developers/contributing.rst

```diff
@@ -810,6 +810,93 @@ To test code coverage, you need to install the `coverage
+that they don't introduce performance regressions. Scikit-learn uses
+`asv benchmarks <https://github.com/airspeed-velocity/asv>`_ to monitor the
+performance of a selection of common estimators and functions. The benchmark
+suite can be found in the `scikit-learn/asv_benchmarks` directory.
+
+To use all features of asv, you will need either `conda` or `virtualenv`. For
+more details please check the `asv installation webpage
+<https://asv.readthedocs.io/en/latest/installing.html>`_.
+
+First of all you need to install the development version of asv::
+
+  pip install git+https://github.com/airspeed-velocity/asv
+
+and change your directory to `asv_benchmarks/`::
+
+  cd asv_benchmarks/
+
+The benchmark suite is configured to run against your local clone of
+scikit-learn. Make sure it is up to date::
+
+  git fetch upstream
+
+In the benchmark suite, the benchmarks are organized following the same
+structure as scikit-learn. For example, you can compare the performance of a
+specific estimator between master and the branch you are working on::
+
+  asv continuous -b LogisticRegression upstream/master HEAD
+
+You can also specify a whole module to benchmark::
+
+  asv continuous -b linear_model upstream/master HEAD
+
+You can replace `HEAD` by any branch. By default it will only report the
+benchmarks that have change by at least 10%. You can control this ratio with
+the `-f` flag.
+
+To run the full benchmark suite, simply remove the `-b` flag ::
+
+  asv continuous upstream/master HEAD
+
+However this can take up to two hours. The `-b` flag also accepts a regular
+expression for a more complex subset of benchmarks to run. The command uses
+conda by default for creating the benchmark environments. If you want to use 
+virtualenv instead, use the `-E` flag::
+
+  asv continuous -E virtualenv upstream/master HEAD
+
+To run the benchmarks without comparing to another branch, use the `run`
+command::
+
+  asv run -b linear_model HEAD^!
+
+You can also run the benchmark suite using the version of scikit-learn already
+installed in your current Python environment::
+
+  asv run --python=same
+
+It's particulary useful when you installed scikit-learn in editable mode. By
+default the results are not saved when using an existing installation. To save
+the results you must specify a commit hash::
+
+  asv run --python=same --set-commit-hash=<commit hash>
+
+Benchmarks are saved and organized by machine, environment and commit. To see
+the list of all saved benchmarks::
+
+  asv show
+
+and to see the report of a specific run::
+
+  asv show <commit hash>
+
+The benchmark suite supports additional configurable options which can be set
+in the `benchmarks/config.json` configuration file. For example, the benchmarks
+can run for a provided list of values for the `n_jobs` parameter.
+
+More information on how to write a benchmark and how to use asv can be found in
+the `asv documentation <https://asv.readthedocs.io/en/latest/index.html>`_.
+
```

Comment:
Should we finally describe what contributors should do once they have run some benchmarks? i.e. where to retrieve the results, and the preferred way to communicate them in the PR / issue?

Reply:
the how to retrieve the results is already described just above. I just added that one should report the results on the github pull request. Is that what you had in mind ?